mantle/cmd/kola/testiso.go

```diff
@@ -123,17 +123,33 @@ func newBaseQemuBuilder() *platform.QemuBuilder {
 	return builder
 }
 
-func newQemuBuilder() *platform.QemuBuilder {
+func newQemuBuilder(isPXE bool) *platform.QemuBuilder {
 	builder := newBaseQemuBuilder()
 	sectorSize := 0
 	if kola.QEMUOptions.Native4k {
 		sectorSize = 4096
 	}
-	builder.AddPrimaryDisk(&platform.Disk{
-		Size: "12G", // Arbitrary
+	primaryDiskAdded := false
+	if system.RpmArch() == "s390x" {
+		// For s390x PXE installs the network device has the bootindex of 1.
```

Comment:
Hmm...like this is something hardcoded into qemu or the firmware or something?

I am trying to get my head around how this works; is it basically that the s390x firmware has builtin logic around "use the disk first if it's bootable, otherwise network" and our adjusting the bootorder breaks it?

Reply:
i am sure it is something like that..i try without the bootindex and it looks for a bootable disk. I have been tried to get a hold of someone from the s390x world to confirm but no one seems to know.

Reply:
yup..found sparse doc here: https://wiki.qemu.org/Features/S390xNetworkBoot